common/djangoapps/student/views.py

```diff
@@ -2297,8 +2302,14 @@ def _get_course_programs(user, user_enrolled_courses):  # pylint: disable=invali
     Returns:
         Dictionary response containing programs or {}
     """
-    course_programs = get_course_programs_for_dashboard(user, user_enrolled_courses)
+    # if cache config is enabled then get the response from cache first.
+    if is_cache_enabled_on_dashboard_programs():
+        programs_data = get_programs_data_from_cache()
```

Comment:
@ahsan-ul-haq I don't think this approach can work, because the data that would be cached here has already been filtered with respect to a specific user's course runs.

The intent of the ticket calls for caching the Programs API response (retrieved here https://github.com/edx/edx-platform/blob/awais786/ECOM-2832-cache-programs-data/openedx/core/djangoapps/programs/views.py#L40-L54).  The caching should happen inside of the `get_course_programs_for_dashboard` function.